From bb091612b29d37ccfe9f82bf86f919a2b8e25397 Mon Sep 17 00:00:00 2001
Message-Id: <bb091612b29d37ccfe9f82bf86f919a2b8e25397.1421830309.git.chang-joon.lee@intel.com>
In-Reply-To: <49dde4951ccd12ce81e60aed3b24793e0ac12fed.1421830308.git.chang-joon.lee@intel.com>
References: <49dde4951ccd12ce81e60aed3b24793e0ac12fed.1421830308.git.chang-joon.lee@intel.com>
From: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Date: Wed, 22 Oct 2014 14:43:27 +0530
Subject: [PATCH 14/17] REVERTME [VPG]: drm/i915: if CHV mipi port C, reset
 display once for cmd mode

There is a hw bug that MIPI Port C reg cannot be read before CHT C0
stepping. Hence do a display reset once for command mode enabling
to switch dsi controller from video to cmd mode.

REVERTME: This patch can be reverted once everyone in Intel has
started using CHT S0C stepping >= C0.

Issue: GMINL-1644
Change-Id: I281f7e5873d6b2f90fb81f50c31c32ba6b913e42
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.c |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index f750b83..58a6218 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -440,7 +440,14 @@ static void intel_dsi_port_disable(struct intel_encoder *encoder)
 
 	dev_priv->video_disabled = false;
 
-	if (is_cmd_mode(intel_dsi) &&
+	if (is_cmd_mode(intel_dsi) && pipe && IS_CHERRYVIEW(dev_priv->dev)
+						&& STEP_TO(STEP_B3)) {
+		static int once_done;
+		if (once_done == false) {
+			once_done = true;
+			dev_priv->video_disabled = true;
+		}
+	} else if (is_cmd_mode(intel_dsi) &&
 			(I915_READ(MIPI_PORT_CTRL(pipe)) & DPI_ENABLE))
 		dev_priv->video_disabled = true;
 
-- 
1.7.9.5

